src/components/organisms/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,118 @@
+import { CSSProperties, Dispatch, SetStateAction, useState } from 'react'
+import classes from './styles.module.scss'
+import classNames from 'classnames'
+import {
+  ExpandedState,
+  getExpandedRowModel,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  PaginationState,
+  getPaginationRowModel,
+  TableMeta,
+  Row,
+  RowData,
+  ColumnDef,
+} from '@tanstack/react-table'
+import { Root } from '@radix-ui/react-form'
+import Container from 'components/atoms/Container/Container'
+import TablePagination from 'components/organisms/TablePagination/TablePagination'
+import TableHeaderGroup from 'components/organisms/TableHeaderGroup/TableHeaderGroup'
+
+export enum TableSizeTypes {
+  L = 'l',
+  M = 'm',
+  S = 's',
+}
+
+type DataTableProps<TData extends RowData> = {
+  data: TData[]
+  columns: ColumnDef<TData>[]
+  tableSize: string
```

Comment:
**[blocker]** Let's use TableSizeTypes when defining types, this way we ensure that only valid values from the TableSizeTypes enum can be assigned to it.